bin/processGsRawData.sh

```diff
@@ -0,0 +1,650 @@
+
+		echo "DEBUG:    Found _flowcell ............... = ${_flowcell}"
+		echo "DEBUG:    Found _runID .................. = ${_runID}"
+	else
+		echo "FATAL: Failed to parse required meta-data values from ID of first read of ${_runPrefix}/${_firstFastQ}"
+		exit 1
+	fi
+
+module load ngs-utils/"${NGS_UTILS_VERSION}"
+module list
+
+cd "${_runPrefix}"
+
+renameFastQs.bash -s "${_sequencingStartDate}" -f "${_flowcell}_${_runID}"'*'
+
+if [ ${?} -eq 0 ]
+then
+	touch "${_controlFileBase}.renameFastQs.finished"
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "renameFastQs finished."
+else
+	touch "${_controlFileBase}.renameFastQs.failed"
+	log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '0' "renameFastQs failed."
+fi
+
+}
+
+function mergeSamplesheetPerProject() {
+	
+	local _run="${1}"
+	local _sampleSheet="${TMP_ROOT_DIR}/Samplesheets/archive/${_run}.${SAMPLESHEET_EXT}"
+	local _controlFileBase="${TMP_ROOT_DIR}/logs/${_run}/${_run}.${SCRIPT_NAME}"
+	local _controlFileFinished="${_controlFileBase}.mergeSamplesheetPerProject.finished"
+	local _logFile="${_controlFileBase}.log"
+	local _processSoFarSoGood='false'
+	
+	if [[ -e "${_controlFileFinished}" ]]
+	then
+		log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Found ${_controlFileFinished} -> Skipping merging ${_run}."
+		return
+	elif [[ ! -e "${_controlFileBase}.sanityChecking.finished" || ! -e "${_controlFileBase}.renameFastQs.finished" ]]
+	then
+		log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '0' "Not Found ${_controlFileBase}.sanityChecking.finished or ${_controlFileBase}.renameFastQs.finished"
+		touch "${_controlFileBase}.mergeSamplesheetPerProject.failed"
+	else
+		log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "No ${_controlFileFinished} present -> Merge samplesheets for ${_run}..."
+		_processSoFarSoGood='true'
+		
+	fi
+	
+
+#
+# combine GS samplesheet with inhouse samplesheets.
+python NGS_Automated/bin/createInhouseSamplesheetFromGS.py \
+--GenomeScanInputDir "${TMP_ROOT_DIR}/${_run}/" \
+--logfile "${_controlFileBase}.log" \
+--samplesheetNewDir "${TMP_ROOT_DIR}/Samplesheets/new/" \
+--samplesheetOutputDir "${TMP_ROOT_DIR}/Samplesheets/"
+
+
+	declare -a _sampleSheetColumnNames=()
+	declare -A _sampleSheetColumnOffsets=()
+	local      _projectFieldIndex
+	declare -a _projects=()
+	
+	cd "${TMP_ROOT_DIR}/${_run}/" 
+	local _runDir=$(find * -type d | grep -P '[0-9]*_[A_Za-z0-9]*_*[0-9]*')
+	cd -
+	_gsSamplesheet=$(ls ${TMP_ROOT_DIR}/${_run}/CSV_UMCG_*.csv)
+	IFS="${SAMPLESHEET_SEP}" _sampleSheetColumnNames=($(head -1 "${_gsSamplesheet}"))
+	for (( _offset = 0 ; _offset < ${#_sampleSheetColumnNames[@]:-0} ; _offset++ ))
+	do
+		_sampleSheetColumnOffsets["${_sampleSheetColumnNames[${_offset}]}"]="${_offset}"
+	done
+	_projectFieldIndex=$((${_sampleSheetColumnOffsets['Sample_ID']} + 1))
+	IFS=$'\n' _projects=($(tail -n +2 "${_gsSamplesheet}" | cut -d "${SAMPLESHEET_SEP}" -f "${_projectFieldIndex}" | sort | uniq ))
+	
+	#
+	# Cat samplesheets projects to _runDir.csv.
```

Comment:
This can be done without cd back and forth en maybe with a cp instead of a cat ;)

Reply:
done